### PR TITLE
Traction Helm Chart Refresh

### PR DESCRIPTION
**Pull request by i5okie**
# Helm Chart Refresh

- Move environment-specific values files out of the `traction` and `tenant-ui` chart directories to `deploy` directory.
- Update GHA workflows to reflect the changes
- Remove image version/buildtime metadata and resources in helm charts

## Traction
- Refactor helm chart templates, helpers, values
- Add autoscaling enabled via a flag in values
- Add `ingress` enabled via a flag in values. This allows the chart to be installed on a common Kubernetes cluster.
- Disable creating Openshift `route` objects by default for the above reason
- Simplify Acapy configuration by generating configmap with `argfile.yml` and `plugin-config.yml` customizable in Values file
- Improve secrets generation (Generated once, reused on `helm upgrade`)
- Allow Postgres configuration and credentials to be passed via existing secrets or individual values
- Add `max_connections = 400` to postgres configuration (Addresses https://github.com/bcgov/traction/issues/558)
- Annotate values file with metadata
- Generate `README.md` with Bitnami [Readme Generator For Helm](https://github.com/bitnami-labs/readme-generator-for-helm)

## Tenant-UI
- Rename chart
- Refactor helm chart templates, helpers, values
- Use `bitnami/common` helm library chart
- Add `ingress` enabled via a flag in values. This allows the chart to be installed on a common Kubernetes cluster.
- Disable creating Openshift route objects by default for the above reason
- Create an `innkeeper` secret from provided existing secret, or by providing the individual values. (Generated once, reused on `helm upgrade`)
- Annotate values file with metadata
- Generate `README.md` with Bitnami [Readme Generator For Helm](https://github.com/bitnami-labs/readme-generator-for-helm)

### CODE REVIEW

File: charts/traction/templates/acapy/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+           "-c",
+           "{{ include "acapy.registerLedger" . }}"
+          ]
+          env:
+            - name: ACAPY_WALLET_SEED
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "acapy.fullname" . }}
+                  key: seed
+      containers:
```
Comment (esune) [inline charts/traction/templates/acapy/deployment.yaml:48]: @usingtechnology is this container still needed? I assume it is what was being used to register DIDs on the ledger, but we switched to using the endorser request instead - correct?

File: deploy/tenant-ui/values-int.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+fullnameOverride: "int-tenant-ui"
```
Comment (esune) [inline deploy/tenant-ui/values-int.yaml:1]: I assume this file will go once the new deployment structure is implemented, correct?

File: charts/traction/templates/acapy/deployment.yaml (hunk 1 above)
Comment (i5okie) [reply charts/traction/templates/acapy/deployment.yaml:48]: I wasn't aware of the switch. Was just moving it out of here https://github.com/bcgov/traction/blob/a3e68e952ad1baf2489377f06f4790054a9a7beb/charts/traction/templates/acapy_deployment.yaml#L39 into its own container.

Comment (esune) [reply charts/traction/templates/acapy/deployment.yaml:48]: Np, just leave it there for now - wanted to confirm with @usingtechnology so we can pull this out in the nxt iteration

Comment (usingtechnology) [reply charts/traction/templates/acapy/deployment.yaml:48]: i would assume it is still needed (this is a new layout for me)... unless you are starting on multiledger? in which case i guess you wouldn't want to register? that API that is called may not be valid for all ledgers anyway

**Review by esune — APPROVED** [review]
👍🏻

**Pull request merged**